### PR TITLE
Suggestion

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please consider.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java [hunk 1]
```diff
@@ -385,20 +391,19 @@ private boolean isRunningProductionJob(Application application) {
...
+     * same version, and was successfully deployed, in which case it is unnecessary to redeploy it.
      */
-    private boolean isOnAtLeastProductionVersion(Version version, Application application, JobType job) {
-        if ( ! job.isProduction()) return false; // TODO: Throw if not prod, and change name/doc accordingly
-        Optional<ZoneId> zone = job.zone(controller.system());
-        if ( ! zone.isPresent()) return false;
-        Deployment existingDeployment = application.deployments().get(zone.get());
-        if (existingDeployment == null) return false;
-        return existingDeployment.version().isAfter(version) || existingDeployment.version().equals(version);
+    private boolean alreadyDeployed(Version version, Application application, JobType job) {
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java]: Not a good name, but man was it hard to find one!?

**Comment by bratseth** [comment]
I like the name.

You restored the "did actually deploy" condition in DeploymentTrigger, so now we check there BOTH if the job succeeded and if the deployment happened.

I think that's ok but unnecessary, since it should never happen that the job succeeded but the deployment did not go through (as in being recorded by the controller). If that happens we should probably warn at least.

BUT, in addition to the above you also change alreadyDeployed to answer true if EITHER the job succeeded OR the deployment happened. I think that is wrong because it brings us back to the case where we will progress to the next job if the previous deployed the version but failed verification tests.

**Comment by jonmv** [comment]
No, I answer true if EITHER the job succeeded OR a _newer_ deployment has happened.

**Comment by bratseth** [comment]
Ah, ok that makes sense - thanks.

**Pull request merged**